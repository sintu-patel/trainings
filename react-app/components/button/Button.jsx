import React from 'react';

class Button extends React.Component {
	constructor(props) {
		super(props);
		this.state = {
			buttonClass: ''
		};
		this.handleClick = this.handleClick.bind(this);
	}
	handleClick() {
		this.setState({
			buttonClass: 'active'
		});
	}
	render() {
		return (
			<button
				type="button"
				className={`button ${this.state.buttonClass}` }
				onClick={this.handleClick }
			>
				{this.props.text}
			</button>
		);
	}
}
export default Button;