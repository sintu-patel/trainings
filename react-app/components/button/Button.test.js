import React from 'react';
import {mount} from 'enzyme';
import renderer from 'react-test-renderer';
import Button from './Button.jsx';

describe('Button', () => {

	const data = {
		text: 'button text'
	};

	test('Snapshot is equal to previous snapshot', () => {
        const tree = renderer
            .create(<Button {...data} />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
    
    const wrapper = mount(<Button {...data} />);
    test('Button Renders', () => {
        expect(wrapper.find('.button').length).toBe(1);
    });

    test('Button Text is same as in props', () => {
        expect(wrapper.find('.button').text()).toBe('button text');
    });

    test('Button does not have class active on load', () => {
        expect(wrapper.find('.button').hasClass('active')).toBeFalsy();
    });

    test('Button does have class active if click on Button', () => {
    	wrapper.find('.button').simulate('click');
        expect(wrapper.find('.button').hasClass('active')).toBeTruthy();
    });
});
