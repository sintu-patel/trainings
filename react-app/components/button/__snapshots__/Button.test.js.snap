// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Snapshot is equal to previous snapshot 1`] = `
<button
  className="button "
  onClick={[Function]}
  type="button"
>
  button text
</button>
`;
